File: LeetCode/problems/657-robot-return-to-origin/657-robot-return-to-origin.js
/**
 * @param {string} moves
 * @return {boolean}
 */
var judgeCircle = function(moves) {
    let X = 0;
    let Y = 0;
    for(let chr of moves){
        if(chr == "L"){X--;}
        if(chr == "R"){X++;}
        if(chr == "U"){Y--;}
        if(chr == "D"){Y++;}
    }
    return X == 0 && Y == 0;
};

File: LeetCode/problems/657-robot-return-to-origin/657-robot-return-to-origin.c
bool judgeCircle(char* moves) {
    short X = 0;
    short Y = 0;
    for(char *p = moves; *p; p++){
        if(*p == 'L'){X--;}
        if(*p == 'R'){X++;}
        if(*p == 'U'){Y--;}
        if(*p == 'D'){Y++;}
    }
    return X == 0 && Y == 0;
}
